psrelease.psm1

```diff
@@ -58,6 +58,23 @@ function Get-Destination
     return $env:TEMP
 }
 
+function Get-TempFolder
+{
+    $tempPath = $env:TEMP
+    if($env:AGENT_TEMPDIRECTORY)
+    {
+        $tempPath = $env:AGENT_TEMPDIRECTORY
+    }
+
+    $tempFolder = Join-Path -Path $tempPath -ChildPath ([System.IO.Path]::GetRandomFileName())
```

Comment:
You are not supposed to use temp on VSTS machines as they are not reset to a base image between builds.  Only `env:AGENT_TEMPDIRECTORY` is cleaned.  I'm falling back to `$env:TEMP` when it doesn't exist for local testing (changed to `[System.IO.Path]::GetTempPath()`)

Added a comment to clarify why.